### DISCUSSION TITLE
Watch flag for the `generate` command

### DISCUSSION THREAD

**Discussion by lorypelli** [a-h/templ#995] [Ideas]
Add it or make the command automatically run on every go run

**Comment by lorypelli** [comment]
![image](https://github.com/user-attachments/assets/bfc193cc-96e4-4dda-bbc8-96034420bede)
I added this to the `Makefile` and now it works, but the thing I hate is that since it's a CLI command, I need to install it globally with `go install github.com/a-h/templ/cmd/templ`

**Comment by pharrisee** [comment]
No need to install (although I would), you can always use something like:

```sh
go run github.com/a-h/templ/cmd/templ@latest generate
```

If you use VSCode it may be worth you looking into devcontainers as you can encapsulate the dev environment per project.

https://containers.dev/ 
https://code.visualstudio.com/docs/devcontainers/containers

Here's an example devcontainer config you might use for dev purposes with templ:

```json
{
        "name": "Go",
        "image": "mcr.microsoft.com/devcontainers/go:1-1-bookworm",
        "features": {
                "ghcr.io/jungaretti/features/make:1": {},
                "ghcr.io/devcontainers-extra/features/go-task:1": {},
                "ghcr.io/devcontainers-extra/features/temporal-cli:1": {}
        }
}
```

The config adds 3 'features' for templ, task and make and installs them ready to use within the container.  The base container already contains go and you could easily add tailwind as another feature if required.

**Comment by a-h** [comment]
`templ generate` already has a `--watch` flag to watch for changes to templ files, and a `--cmd` flag to run afterwards.

Perhaps you could clarify what you're looking for?

**Reply by lorypelli** [reply]
That one actually, even if it doesn't work perfectly and I need to full restart quite a lot

**Reply by catgoose** [reply]
> That one actually, even if it doesn't work perfectly and I need to full restart quite a lot

This is how I do it:

.air.toml (see: https://templ.guide/developer-tools/live-reload)

```
...
[build]
...
  cmd = "templ generate --notify-proxy && go build -o ./tmp/main ."

```

Makefile

```
PROXY_HOST = localhost
PROXY_URL := http://$(PROXY_HOST):1337
BIN_PATH := ./bin
PUBLIC_SOURCE_DIR := web/assets/public
PUBLIC_CSS_DIR := $(PUBLIC_SOURCE_DIR)/css
TAILWIND_CMD := $(BIN_PATH)/tailwindcss -i web/styles/input.css -o $(PUBLIC_CSS_DIR)/tailwind.css -m

# Watch mode
templ-watch:
	templ generate -watch -proxy=$(PROXY_URL) -proxybind=$(PROXY_HOST)

air:
	air

tailwind-watch:
	$(TAILWIND_CMD) -w

watch:
	$(MAKE) -j3 templ-watch air tailwind-watch
```

when I run `make watch` I do get the error:

`(✗) Command failed: Post "http://127.0.0.1:7331/_templ/reload/events": dial tcp 127.0.0.1:7331: connect: connection refused`

But it doesn't seem to affect anything.
